### ISSUE TITLE
flycheck rustic-clippy clippy installation guide is wrong

### ISSUE THREAD

**Issue by ncaq** [brotzeit/rustic#18]
I see

~~~
Suspicious state from syntax checker rustic-clippy: Flycheck checker rustic-clippy returned non-zero exit code 1, but its output contained no errors: error: 'cargo-clippy' is not installed for the toolchain 'nightly-x86_64-unknown-linux-gnu'
To install, run `rustup component add clippy`
~~~

I do

~~~
% rustup component add clippy
info: component 'clippy' for target 'x86_64-unknown-linux-gnu' is up to date
~~~

but, This is not work.

I do

~~~
% rustup component add --toolchain nightly clippy
info: downloading component 'clippy'
info: installing component 'clippy'
~~~

This is work.

I can not search error message,
And I can not fix error message.

**Comment by brotzeit** [comment]
Hmmm. I have to take a look at this.

**Comment by mikhail-krainik** [comment]
@brotzeit is there any updates? I also have such a problem after update rust nightly version.

**Comment by brotzeit** [comment]
Sorry I'm currently working mostly on remacs, but I will take a look at the open rustic issues the next days.

**Comment by brotzeit** [comment]
I don't know where this error message is from. This must be some flycheck magic or I don't know. I have to take a deeper look at the code I copied from flycheck-rust.

**Comment by mikhail-krainik** [comment]
> I don't know where this error message is from. This must be some flycheck magic or I don't know. I have to take a deeper look at the code I copied from flycheck-rust.

In my case, this happened when I updated rust night to 1.32.0 and stable to 1.31.0. I hope this information will be helpful.

**Comment by brotzeit** [comment]
I'm on it. Unfortunately I don't know much about flycheck.

**Comment by brotzeit** [comment]
This installation instructions are coming from cargo, so this can't be fixed in rustic.

Are there any other issues with flycheck ?

**Comment by mikhail-krainik** [comment]
> This installation instruction are coming from cargo, so this can't be fixed in rustic.
> 
> Are there any other issues with flycheck ?

Not clear what I should do next, because I already updated cargo-clippy on the latest version.
> Suspicious state from syntax checker rustic-clippy: Flycheck checker rustic-clippy returned non-zero exit code 1, but its output contained no errors: error: 'cargo-clippy' is not installed for the toolchain 'nightly-x86_64-unknown-linux-gnu'
To install, run `rustup component add clippy`
> Try installing a more recent version of rustic-clippy, and please open a bug report if the issue persists in the latest release.  Thanks!

And what it means?

> more recent version of rustic-clippy

I don't have this module just rustic-cargo-clippy.

**Comment by brotzeit** [comment]
> Try installing a more recent version of rustic-clippy, and please open a bug report if the issue persists in the latest release. Thanks!

This is the message from flycheck.

Does `rustup component add --toolchain nightly clippy` work for you ?

**Comment by brotzeit** [comment]
Did you try it with a different toolchain ? clippy isn't available for every toolchain. There was a site were you can see which toolchain contains clippy etc. but I can't find it.

**Comment by brotzeit** [comment]
FYI this isn't flycheck related.

**Comment by mikhail-krainik** [comment]
In my system already installed some clippy
```
$ cargo clippy --version
clippy 0.0.212 (754b4c0 2018-11-23)
```
Maybe there is a problem with different version whith can't communicate with each other. I don't know.

**Comment by brotzeit** [comment]
Uninstall it and install it with the version from above. Worked for me...

**Comment by brotzeit** [comment]
I close this as it isn't rustic related.

**Comment by zslayton** [comment]
I found this issue by googling for the error text in the description, so I thought I'd leave a comment with the solution that worked for me. 

### TL;DR

`rustic-clippy` was failing because I didn't have enough RAM on my machine for it to build the project in the background.

### Details

I had installed Doom emacs on a VSP with a small amount of RAM. The first time I opened a `.rs` file, a few seconds of intense CPU work happened before the echo area showed the message:

> Suspicious state from syntax checker rustic-clippy: Flycheck checker rustic-clippy returned non-zero exit code 1, but its output contained no errors: error: 'cargo-clippy' is not installed for the toolchain 'nightly-x86_64-unknown-linux-gnu'
To install, run `rustup component add clippy`

This was peculiar as I was using the `stable` toolchain (v1.49). Clippy was already installed, so I ran `cargo clippy` in the shell (rather than via `rustic-mode` in emacs). It worked; clippy began trying to build the project's dependencies. It made it a fair ways down the dependency tree before failing again.

Running `dmesg` showed:

```
[628353.797161] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-0.slice/session-23.scope,task=clippy-driver,pid=198570,uid=0
[628353.797209] Out of memory: Killed process 198570 (clippy-driver) total-vm:560108kB, anon-rss:245592kB, file-rss:0kB, shmem-rss:0kB, UID:0 pgtables:984kB oom_score_adj:0
```

The issue was insufficient memory: if emacs (and some other things) were running on the VPS, it didn't have enough memory to compile heavyweight packages like `syn` and `http`.

I stopped everything that was running and ran `cargo clippy`. It finished building successfully. I re-opened my `.rs` file, and everything worked as expected.
